src/amo/components/Collection/index.js

```diff
@@ -385,10 +385,6 @@ export const mapStateToProps = (
   let hasEditPermission = false;
 
   const collection = getCurrentCollection(state.collections);
-  if (collection && currentUser) {
```

Comment:
We cannot remove all of this code, we need to only remove some of it, because we still need `hasEditPermission` to be `true` if `collection.authorId === currentUser.id`.